src/Microsoft.AspNetCore.Authentication.OAuth/Events/OAuthCreatingTicketContext.cs

```diff
@@ -13,58 +13,48 @@ namespace Microsoft.AspNetCore.Authentication.OAuth
     /// <summary>
     /// Contains information about the login session as well as the user <see cref="System.Security.Claims.ClaimsIdentity"/>.
     /// </summary>
-    public class OAuthCreatingTicketContext : BaseAuthenticationContext
+    public class OAuthCreatingTicketContext : ResultContext<IOAuthHandler>
     {
         /// <summary>
         /// Initializes a new <see cref="OAuthCreatingTicketContext"/>.
         /// </summary>
+        /// <param name="handler">The authentication handler.</param>
         /// <param name="ticket">The <see cref="AuthenticationTicket"/>.</param>
         /// <param name="context">The HTTP environment.</param>
         /// <param name="scheme">The authentication scheme.</param>
-        /// <param name="options">The options used by the authentication middleware.</param>
         /// <param name="backchannel">The HTTP client used by the authentication middleware</param>
         /// <param name="tokens">The tokens returned from the token endpoint.</param>
         public OAuthCreatingTicketContext(
+            IOAuthHandler handler,
```

Comment:
Yeah given how little time we have, maybe we should just do the rename + related cleanup, but not push generics into the base events.  Keep the existing BaseXyzContext's that provide the correctly typed Options property.  But we can still push the Scheme down.